### PR TITLE
Use TLS friendly addresses in inspect output [full ci]

### PR DESCRIPTION
**Pull request by hickeng**
Attempts to determine the appropriate address to use based on the configured certificates.
This is necessary because with certificate validation happening you have to use the appropriate name/IP to connect to the system.

The primary effect here is that the vicadmin URL will be either IP or FQDN as needed

Full CI because anything that changes vic-machine output is fraught with danger.

Towards #2726

### CODE REVIEW

**Comment by hickeng** [comment]
@dougm @gigawhitlocks seeing as you've both been messing with cert stuff you're good people to review.

**Comment by dougm** [comment]
lgtm

File: lib/install/management/dispatcher.go [hunk 1]
```diff
@@ -211,41 +211,87 @@ func (d *Dispatcher) CollectDiagnosticLogs() {
...
-		log.Debugf("Checked %s, no match for host cert", n)
+		candidateIPs = append(candidateIPs, ips...)
 	}
 
-	// no viable address
-	return ip, errors.New("unable to determine viable address")
+	// always add all the altname IPs - we're not checking for connectivity
+	candidateIPs = append(candidateIPs, cert.IPAddresses...)
+
+	return addrToUse(candidateIPs, cert, cas)
```
Comment (hmahmood) [inline lib/install/management/dispatcher.go]: So this resolves the ips back to dns names, while we did the reverse in this function. Why?

File: cmd/vic-machine/create/create.go [hunk 2]
```diff
@@ -861,6 +865,17 @@ func (c *Create) generateCertificates(ca bool) ([]byte, *certificate.KeyPair, er
 		log.Warnf("Failed to stash client certificate for later application level validation: %s", err)
 	}
 
+	// if openssl is present, try to generate a browser friendly pfx file
```
Comment (caglar10ur) [inline cmd/vic-machine/create/create.go]: What is browser friendly cert? Can we add a comment explaining what's going on here?

File: lib/install/management/appliance.go [hunk 3]
```diff
@@ -678,31 +676,51 @@ func (d *Dispatcher) CheckDockerAPI(conf *config.VirtualContainerHostConfigSpec,
...
-			pool := x509.NewCertPool()
-			if !pool.AppendCertsFromPEM(conf.CertificateAuthorities) {
-				log.Debug("Unable to load CAs in config, if any")
-			}
+			// if tlsverify was configured at all then we must verify the remote
+			tr.TLSClientConfig.InsecureSkipVerify = false
 
-			// tr.TLSClientConfig.ClientCAs = pool
-			tr.TLSClientConfig.RootCAs = pool
+			func() {
```
Comment (caglar10ur) [inline lib/install/management/appliance.go]: why do we have this? I don't see any defer's here so I guess we could remove the func(){ / {() parts, no?

File: cmd/vic-machine/create/create.go (hunk 2 above)
Comment (hickeng) [inline cmd/vic-machine/create/create.go]: Added some additional comments.

**Comment by caglar10ur** [comment]
LGTM

**Comment by hickeng** [comment]
poke

File: lib/install/management/dispatcher.go (hunk 1 above)
Comment (hickeng) [inline lib/install/management/dispatcher.go]: We resolve the DNS names to IPs as a filter for which are valid at the location vic-machine is being run. We then use that set of possible IPs along with the certificates to see which addresses pass TLS validation.

**Pull request merged**